src/registries/npm-registry.js

```diff
@@ -74,21 +76,50 @@ export default class NpmRegistry extends Registry {
   }
 
   getRequestUrl(registry: string, pathname: string): string {
-    const isPathnameUrl = pathname.match(/^https?:/);
+    const isUrl = /^https?:/.test(pathname);
 
-    if (isPathnameUrl) {
+    if (isUrl) {
       return pathname;
     } else {
       return url.resolve(registry, pathname);
     }
   }
 
+  getPortOrDefaultPort(port: ?string, protocol: ?string): ?string {
+    if (protocol === 'https:' && port === DEFAULT_HTTPS_PORT.toString()) {
+      return null;
+    }
+    if (protocol === 'http:' && port === DEFAULT_HTTP_PORT.toString()) {
+      return null;
+    }
+    return port;
+  }
+
+  isRequestToRegistry(requestUrl: string, registry: string): boolean {
+    const requestParsed = url.parse(requestUrl);
+    const registryParsed = url.parse(registry);
+    const requestHost = requestParsed.hostname || '';
+    const registryHost = registryParsed.hostname || '';
+    const requestPort = this.getPortOrDefaultPort(requestParsed.port, requestParsed.protocol);
+    const registryPort = this.getPortOrDefaultPort(registryParsed.port, registryParsed.protocol);
+    const requestPath = requestParsed.path || '';
+    const registryPath = registryParsed.path || '';
+    const customHostSuffix = this.getRegistryOrGlobalOption(registry, 'custom-host-suffix');
+
+    return (
+      requestHost === registryHost &&
+      requestPort === registryPort &&
+      (requestPath.startsWith(registryPath) ||
+        // For some registries, the package path does not prefix with the registry path
+        (typeof customHostSuffix === 'string' && customHostSuffix.length > 0 && requestHost.endsWith(customHostSuffix)))
```

Comment:
Oh, makes sense. Well, may be you can do
```
const customHostSuffix = String(this.getRegistryOrGlobalOption(registry, 'custom-host-suffix') || '');
```

So it is always a string? The only downside is if someone someone sets this to `null` or something, it will be converted into the string `'null'`. Not too concerned about that since AFAIK `typeof` checks are expensive.